app/org/sagebionetworks/bridge/play/controllers/ScheduleController.java

```diff
@@ -47,8 +48,12 @@ public Result getSchedulesV3() throws Exception {
         List<SchedulePlan> plans = schedulePlanService.getSchedulePlans(clientInfo, studyId);
         List<Schedule> schedules = Lists.newArrayListWithCapacity(plans.size());
         for (SchedulePlan plan : plans) {
-            Schedule schedule = plan.getStrategy().getScheduleForUser(session.getStudyIdentifier(), plan, session.getUser());
-            schedules.add(schedule);
+            ScheduleContext context = new ScheduleContext.Builder()
+                    .withUser(session.getUser()).withClientInfo(clientInfo).build();
```

Comment:
ScheduleContext should be constructed outside the loop, since it doesn't vary with the loop.